walkbox.go

```diff
@@ -239,25 +247,68 @@ func (wm *WalkBoxMatrix) FindPath(from, to Position) []*WayPoint {
 	var path []*WayPoint
 	fromf := from.ToPosf()
 	tof := to.ToPosf()
-	current, _ := wm.walkBoxAt(fromf)
-	target, _ := wm.walkBoxAt(tof)
+	current, _ := wm.walkBoxIndex(fromf)
+	target, _ := wm.walkBoxIndex(tof)
+	currentWp := &WayPoint{Walkbox: wm.walkBoxes[current], Position: from}
 
-	path = append(path, &WayPoint{Walkbox: wm.walkBoxes[current], Position: from})
+	path = append(path, currentWp)
 	for current != target {
 		next := wm.nextWalkBox(current, target)
 		if next == InvalidWalkBox {
 			break
 		}
 		nextPosition := wm.closestPositionToWalkBox(fromf, next)
-		path = append(path, &WayPoint{Walkbox: wm.walkBoxes[next], Position: nextPosition.ToPos()})
+		nextWp := &WayPoint{Walkbox: wm.walkBoxes[next], Position: nextPosition.ToPos()}
+		path = append(path, append(currentWp.interpolate(nextWp), nextWp)...)
 		current = next
+		currentWp = nextWp
 		fromf = nextPosition
 	}
 
 	path = append(path, &WayPoint{Walkbox: wm.walkBoxes[current], Position: wm.closestPositionOnWalkBox(current, tof).ToPos()})
 	return path
 }
 
+// interpolate generates intermediate waypoints between a starting and ending waypoint.
```

Comment:
It is a provisional solution to allow smooth transition between different scales but the problem is, with more points, actor seems to "blink" changing direction  etc,..

I need more time to think a better solution